### DISCUSSION TITLE
Not working with the Saferpay WooCommerce Gateway Payment Method

### DISCUSSION THREAD

**Discussion by kra2010** [woocommerce/woocommerce#45054] [WooCommerce, Blocks, and Site Editing]
## Describe the bug
Woocommerce blocks error:
There are no payment methods available. This may be an error on our side. Please contact us if you need any help placing your order.

## To reproduce

Installed payment provider (plugin woocommerce): https://www.webshopextension.com/de/saferpay-woocommerce
this works fine with the normal woocommerce checkout, but woocommerce blocks is not able to recognise it.

## Expected behavior
Payments should be possible :) (MasterCard, VISA, Apple Pay is working with the woocommerce_checkout

## Screenshots
There are no payment methods available. This may be an error on our side. Please contact us if you need any help placing your order.

### WordPress (please complete the following information):
* Core version: 5.9.8
* Gutenberg version (if installed): -
* WooCommerce version: 5.9.0
* WooCommerce Blocks version: 6.2.0
* Site language: English. 
* www.charmingc.ch
* Any other plugins installed: Saferpay Woocommerce Gateway

### Desktop (please complete the following information):
* OS: macOs
* Browser chrome, safari
* Version [e.g. 22]

### Smartphone (please complete the following information):
* Device: [e.g. iPhone6]
* OS: [e.g. iOS8.1]
* Browser [e.g. stock browser, safari]
* Version [e.g. 22]

**Comment by bajoski34** [comment]
Hi @kra2010  This is because the third party plugin does not support woocommerce blocks.Please kindly take some time to read this https://developer.woocommerce.com/2021/03/15/integrating-your-payment-method-with-cart-and-checkout-blocks/

**Comment by Aljullu** [comment]
Hi there @kra2010  👋
 
Thanks for taking the time to make us aware that Saferpay WooCommerce Gateway isn't working with the Checkout and Cart blocks. These blocks are still in early access (via the feature plugin) and it's expected that some existing Woo Extensions that integrate with the current checkout flow in WooCommerce core will not work with the blocks.
 
Like @bajoski34 mentioned, currently it's up to payment method extension developers to add support for the new checkout flow offered via the cart and checkout blocks. We have some [documentation on that here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/extensibility/payment-method-integration.md) and you can also point them to [the post bajoski shared](https://developer.woocommerce.com/2021/03/15/integrating-your-payment-method-with-cart-and-checkout-blocks/).
 
Your best course of action right now if you are using a payment method that isn't working with the blocks is to notify the developer of the extension and ask if they can add support.
 
Until then you can keep on top of the [status of the Cart & Checkout Blocks in this documentation](https://docs.woocommerce.com/document/cart-checkout-blocks-support-status/).

**Comment by kra2010** [comment]
Hi @bajoski34 and @Aljullu 

Thank you both for your feedback.
As suggested from you both I forwarded the question to my payment gateway Saferpay.
They are looking into it but confirmed already that it is currently not compatible.

I am technically not good enough to help out further.

Best regards and thanks again!
